CheckSite.py

```diff
@@ -1,194 +1,284 @@
 #!/usr/bin/env python
 
 import json
+# from pandas.io.json import json_normalize
+import threading
 import requests
 import sqlite3
 import pandas as pd
+import datetime
+from abc import ABC, abstractmethod
 
+class Website(ABC, threading.Thread):
+    @abstractmethod
+    def getAPI(self):
+        # obtains the API request from the website and returns a JSON object to parse
+        pass
+    def writeDB(self):
+        # create the DB and put the JSON object into it
+        pass
+    def cleanup(self):
+        #remove all outdated lines from the table in the DB
+        pass
+    def run(self):
+        # thread run functionality
+        pass
+
+
+class DrugShortages(Website):
+    def run(self):
+        first_pull =self.getAPI(start_month="5", start_day="10", end_month="5", end_day="10")
```

Comment:
And maybe we should change the types for the months, so instead of strings they are integers since that's easier to use in the function calls. 

Example:

```
obj.method("1", "2", "3")
```
is more awkward than
```
obj.method(1, 2, 3)
```

Reply:
does the API require them to be strings? On the other hand, I can always turn them to string later 👍

Reply:
Yeah exactly, I think we should cast them as strings inside the method.